test/Main.hs

```diff
@@ -64,5 +67,26 @@ translationPasses =
     [ FrontendDesugar.allDesugar
     ]
 
+contractTests :: FilePath -> IO ()
+contractTests file = do
+  parsed <- readFile file
+  let parsedIdString = (parse (many1 topLevelSN) . removeComments) (encodeUtf32BE parsed)
```

Comment:
I have reason to believe it isn't parsing anything atm. 

The most recent commit using `many1` (which applies the action p one or more times) and `parse` instead of `parseOnly` yields 'Fail', with an empty context `[]` and the result of `Left "string"`:

```
juvix> test (suite: juvix-test)
                   
Fail-- standard library functions

let string = "%prim.string"

let pair = "%Michelson.pair"

-- Identity contract for strings. 

type storage = unit 

-- Default entry point
sig default : string -> storage -> storage
��let default para storage = pair [] storage

-- Initialize storage to ().
sig init-storage : storage
let init-storage = ()
[]"string"
```

When `many` is used, it succeeds but it seems it's because it applies action p *zero* times. From the output, you can see the result is an empty list:
```
juvix> test (suite: juvix-test)
                   
Success[]
```